### PR TITLE
Create a Worker.transform operator.

### PR DESCRIPTION
**Pull request by zach-klippenstein**

### CODE REVIEW

File: kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt [hunk 1]
```diff
@@ -36,6 +37,8 @@ import kotlin.reflect.KClass
 /**
  * Represents a unit of asynchronous work that can have zero, one, or multiple outputs.
  *
+ * [Worker]s are simply [Flow]s that can be compared to determine equivalence.
```
Comment (rjrjr) [inline kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt]: I think you mean _effectively_. :trollface:

File: kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/AiWorkflow.kt [hunk 2]
```diff
@@ -95,12 +96,12 @@ private fun <T : Enum<T>> Random.nextEnum(enumClass: KClass<T>): T {
...
 private fun Worker<Long>.createDirectionTicker(random: Random): Worker<Unit> =
-  Worker.create(key = UUID.randomUUID().toString()) {
-    run()
-        .collect { tick ->
-          if (tick % random.nextInt(2, 5) == 0L) {
-            emit(Unit)
-          }
-        }
+  transform {
+    it.transform { tick ->
```
Comment (rjrjr) [inline kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/AiWorkflow.kt]: Maybe more readable if you s/it/flow/

File: kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt [hunk 3]
```diff
@@ -184,6 +187,11 @@ interface Worker<out T> {
       block: suspend () -> Unit
     ): Worker<Nothing> = TypedWorker(Nothing::class, key, flow { block() })
 
+    /**
+     * Returns a [Worker] that finishes immediately without emitting anything.
+     */
+    fun finished(): Worker<Nothing> = FinishedWorker
```
Comment (rjrjr) [inline kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt]: Unused? Should at least be covered by a smoke test.

Comment (zach-klippenstein) [reply kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt]: Everything needs unit tests here.

Comment (zach-klippenstein) [reply kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt]: It's not unused, see GameWorkflow above, line 105.

I'm wondering if this should be a function though, or we should just expose it as a val. No need to do any casting here so no requirement to have it as a function.

Comment (rjrjr) [reply kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt]: +1 to making it a val

Comment (zach-klippenstein) [reply kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt]: I'm going to keep it a function to be consistent with what Kotlin does: `emptyFlow()` is a function. I'm also going to give it a generic, so you can say `Worker.finished<String>()`.

https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/empty-flow.html

**Comment by zach-klippenstein** [comment]
Added unit tests.

**Pull request merged**